src/modules/api.js

```diff
@@ -0,0 +1,78 @@
+import handleFetchError from './errorHandler.js';
+
+const API_BASE_URL = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/'; // Make sure to import the handleFetchError function from the index.js file correctly
+
+export const createNewGame = async (gameName) => {
+  try {
+    const response = await fetch(`${API_BASE_URL}games/`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        name: gameName,
+      }),
+    });
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+
+    const data = await response.json();
+    const gameId = data.result.split(': ')[1]; // Store the ID of the game
+
+    // Store the game id in local storage
+    localStorage.setItem('gameId', gameId);
+
+    return gameId;
+  } catch (error) {
+    handleFetchError(error);
+    throw error; // Throw the error again to maintain consistency in return values
+  }
+};
+
+export const getGameScores = async () => {
+  try {
+    // Get the game id from local storage
+    const gameId = localStorage.getItem('gameId');
+
+    const response = await fetch(`${API_BASE_URL}games/${gameId}/scores`);
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+
+    const data = await response.json();
+    return data.result;
+  } catch (error) {
+    handleFetchError(error);
+    throw error; // Throw the error again to maintain consistency in return values
+  }
+};
+
+export const postGameScore = async (userName, score) => {
+  try {
+    // Get the game id from local storage
+    const gameId = localStorage.getItem('gameId');
+
+    const response = await fetch(`${API_BASE_URL}games/${gameId}/scores`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        user: userName,
+        score,
+      }),
+    });
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
```

Comment:
- [OPTIONAL] Your app works as expected. :+1: Adding a score and fetching scores from the API works great. But looking at the screenshot below, you can see there is an error on the console. When a add anew score successfully, the error shows on the console. although the user will not see this in the UI, I'll suggest you take a look at this and try to fix it. :+1:  :+1: 

- - Screenshot:
![Screenshot from 2023-06-21 21-08-31](https://github.com/ClaudiaRojasSoto/Leaderboard/assets/107632780/6c18e533-3a40-423b-a4eb-abc8674043ac)

- [OPTIONAL] Also for the optimal user experience, I'll suggest you call the `refreshScores` function every time anew score is added, the user gets to see the score without a need to click on the refresh button each time :smile: :+1:

Reply:
I did it, thank you for your comments, but I still have the error in the console... I will try to fix that this night. Thank you again.